### PR TITLE
feat: Mathjax for formules formatting

### PR DESCRIPTION
**Pull request by vladislavkeblysh**
### Description
- Add Mathjax lib for formatting formules

#### Screenshots before:
| image |
|---|

#### Screenshots after:
| image |
|---|

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @vladislavkeblysh!


This repository is currently maintained by `@codewithemad`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>

**Comment by vladislavkeblysh** [comment]
Has problem with CI checker. Tried to use node version 18.19 and npm 9/10. Problems still unsolved

**Comment by brian-smith-tcril** [comment]
@vladislavkeblysh I tried pulling this locally and running
```sh
nvm use
npm ci
```
and I was able to reproduce the error CI is showing. I then ran
```sh
npm install
```
and then
```sh
npm ci
```
and the error was no longer there.

This makes me think that CI should make it past the current error if you run
```sh
npm install
```
and check in the updated `package-lock.json`

**Comment by vladislavkeblysh** [comment]
@brian-smith-tcril I did the same thing, but the problem persists.

**Comment by brian-smith-tcril** [comment]
@vladislavkeblysh what do you see locally when running
```sh
rm -rf node_modules
nvm use
npm ci
```

**Comment by arbrandes** [comment]
We'll probably want product review, here.  Mathjax support across the platform is somewhat inconsistent, and we'll likely want to figure out a strategy going forward.

**Comment by mphilbrick211** [comment]
@jmakowski1123 - is there a product roadmap ticket for this?

**Comment by jmakowski1123** [comment]
This one needs a roadmap ticket and product review.

**Comment by mphilbrick211** [comment]
> This one needs a roadmap ticket and product review.

Thanks @jmakowski1123 - can you link to the roadmap ticket here (once it's live)?

**Comment by vladislavkeblysh** [comment]
@arbrandes @brian-smith-tcril ready to review

**Comment by mphilbrick211** [comment]
Per @jmakowski1123 this does not need to go through Prod review. All set to merge after review.

**Review by arbrandes — CHANGES_REQUESTED** [review]
The code looks alright, but I'm concerned that this uses a completely different method to load MathJax than the current pattern in the codebase, which is to load MathJax 2.7.5 directly from a CDN.  ([edx-platform example](https://github.com/openedx/edx-platform/blob/master/common/templates/xblock_v2/xblock_iframe.html#L190-L190), [frontend example](https://github.com/openedx/frontend-app-ora/blob/master/src/utils/renderMathJax.js#L67-L67)).

I also don't see an attempt to replicate the MathJax configuration used elsewhere ([edx-platform example](https://github.com/openedx/edx-platform/blob/master/common/templates/mathjax_include.html#L40-L57), [frontend example](https://github.com/openedx/frontend-app-ora/blob/master/src/utils/renderMathJax.js#L43-L57)).  What makes this worse is that apparently there is no single configuration pattern in the codebase.  Has an evaluation been made to check that what configuration is actually necessary?

In other words: how likely is it that loading MathJax via `better-react-mathjax` will result in a rendering that is different than what the author intended?

**Comment by vladislavkeblysh** [comment]
@arbrandes Thanks for pointing that out. I’ve checked both MathJax configurations you referenced.

I’ve updated the configuration in better-react-mathjax to match them:
	•	MathJax v2.7.5 (same as current CDN load)
	•	inlineMath and displayMath definitions from both templates
	•	processEscapes: true and SVG max-width fixes
	•	messageStyle: 'none' and automatic linebreaks for all output types

**Comment by mphilbrick211** [comment]
Hi @vladislavkeblysh! Is this still in progress?

**Comment by mphilbrick211** [comment]
Checking in on this, @arbrandes @vladislavkeblysh

**Comment by arbrandes** [comment]
I have one last request for @vladislavkeblysh besides getting CI to pass: can you please capture a couple of screenshots comparing how this PR renders Mathjax versus the LMS and CMS for the same block?

**Comment by mphilbrick211** [comment]
> I have one last request for @vladislavkeblysh besides getting CI to pass: can you please capture a couple of screenshots comparing how this PR renders Mathjax versus the LMS and CMS for the same block?

Flagging this for you, @vladislavkeblysh!

**Comment by vladislavkeblysh** [comment]
@arbrandes Hi, sorry for the long waiting

PR is updated:

- Fixed CI and updated MathJax config from v2 format (tex2jax, CommonHTML) to v3 (tex, chtml, svg) — the old config was incompatible with better-react-mathjax which loads MathJax 3. 
- Resolved merge conflicts with master. 
- Added comparison screenshots.

lms-student-view-mathjax
ora-grading-mfe-mathjax

**Review by arbrandes — APPROVED** [review]
Alright, I'm convinced.  Wherever there's MathJax under React, we should use this pattern.
